pkg/util/azureclient/mgmt/compute/disks_addons.go

```diff
@@ -20,3 +23,21 @@ func (c *disksClient) DeleteAndWait(ctx context.Context, resourceGroupName strin
 
 	return future.WaitForCompletionRef(ctx, c.Client)
 }
+
+func (c *disksClient) List(ctx context.Context, resourceGroupName string) (result []mgmtcompute.Disk, err error) {
+	page, err := c.DisksClient.ListByResourceGroup(ctx, resourceGroupName)
+	if err != nil {
+		return nil, err
+	}
+
+	for page.NotDone() {
+		result = append(result, page.Values()...)
+
+		err = page.NextWithContext(ctx)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	return result, nil
+}
```

Comment:
It looks like you are doing checks by listing VMs, not disks directly. Does this mean that we can remove this?